smithy-utils/src/main/java/software/amazon/smithy/utils/IoUtils.java

```diff
@@ -365,19 +363,43 @@ public FileVisitResult postVisitDirectory(Path dir, IOException e) throws IOExce
     }
 
     public static void copyDir(Path src, Path dest) {
-        try (Stream<Path> stream = Files.walk(src)) {
-            stream.forEach(source -> copyFile(source, dest.resolve(src.relativize(source))));
+        try {
+            Files.walkFileTree(src, new CopyFileVisitor(src, dest));
         } catch (IOException e) {
             throw new RuntimeException(String.format(
                 "Error copying directory from \"%s\" to \"%s\": %s", src, dest, e.getMessage()), e);
         }
     }
 
-    public static void copyFile(Path source, Path dest) {
-        try {
-            Files.copy(source, dest, REPLACE_EXISTING);
-        } catch (Exception e) {
-            throw new RuntimeException(e.getMessage(), e);
+    private static final class CopyFileVisitor extends SimpleFileVisitor<Path> {
```

Comment:
I surprised this needed to change for this PR?

Reply:
This is related to my earlier change about allowing templates to be copied to nested output directory